src/profiler.ts

```diff
@@ -121,17 +120,19 @@ export class Profiler extends common.ServiceObject {
    * @param delayMillis - milliseconds to wait before polling Stackdriver
    * Profiler server to see if it's time to create a profile and profile.
    */
-  async createAndUploadProfile(delayMillis: number): Promise<void> {
-    this.profilingTimeoutId = setTimeout(async () => {
-      const startCreateMillis = Date.now();
-      const prof = await this.createProfile();
-      await this.profileAndUpload(prof);
-      const endCreateMillis = Date.now();
-      this.createAndUploadProfile(
-          this.config.minTimeBetweenProfilesMillis -
-          (endCreateMillis - startCreateMillis));
-    }, delayMillis);
-    this.profilingTimeoutId.unref();
+  async coordinateProfiling(delayMillis = 0): Promise<void> {
```

Comment:
You don't use the parameter anymore. You will also need to update the JSDoc which no longer reflect what the function does. IMO, the name is not very intuitive either.